pages/finance.tsx

```diff
@@ -16,11 +28,74 @@ function Finance() {
       </div>
     )
 
-  // TODO: add useOrientation using custom hook.
-  const startDate = dateBeforeDays(3)
-  const endDate = dateAfterDays(3)
-
-  return <div>{user ? <FinanceDateHeader startDate={startDate} endDate={endDate} /> : null}</div>
+  return (
+    <div>
+      {user ? (
+        <div>
+          <FinanceDateHeader startDate={values.startDate} endDate={values.endDate} />
+          <div className="text-center block relative w-full bottom-0 m-4">
+            <Button
+              size="large"
+              type="submit"
+              variant="contained"
+              color="primary"
+              className={classes.button}
+              onClick={() =>
+                setValues({
+                  startDate: dateBeforeDays(values.startDate, 30),
```

Comment:
handle months with 31 and 28, 29 days as well. IT should always show 1st of the previous month